app/models/webhook.rb

```diff
@@ -0,0 +1,153 @@
+# == Schema Information
+#
+# Table name: webhooks
+#
+#  id             :integer          not null, primary key
+#  namespace_id   :integer
+#  url            :string(255)
+#  username       :string(255)
+#  password       :string(255)
+#  request_method :integer
+#  content_type   :integer
+#  enabled        :boolean          default("0")
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#
+# Indexes
+#
+#  index_webhooks_on_namespace_id  (namespace_id)
+#
+
+require "base64"
+require "typhoeus"
+require "securerandom"
+require "json"
+require "uri"
+
+# A Webhook describes a kind of callback to an endpoint defined by an URL.
+# Futher parameters are username and password, which are used for basic
+# authentication. The parameters request_method and content_type are limitted
+# to GET and POST, and application/json and application/x-www-form-urlencoded
+# respectively. Webhooks can be enabled or disabled.
+# After a webhook has been triggered with the provided parameters, a
+# WebhookDelivery object is created.
+class Webhook < ActiveRecord::Base
+  include PublicActivity::Common
+
+  enum request_method: ["GET", "POST"]
+  enum content_type: ["application/json", "application/x-www-form-urlencoded"]
+
+  belongs_to :namespace
+
+  has_many :deliveries, class_name: "WebhookDelivery"
+  has_many :headers, class_name: "WebhookHeader"
+
+  validates :url, presence: true, url: true
+
+  # default to http if no protocol has been specified. If unspecified, the URL
+  # validator will fail.
+  before_validation do
+    unless url.nil? || url.strip.empty?
+      self.url = "http://#{url}" unless url.start_with?("http://") ||
+          url.start_with?("https://")
+    end
+  end
+
+  before_destroy do
```

Comment:
You can achieve this by saying that `headers` and `deliveries` depend on this model. If I'm not mistaken, this can be done like this in the `has_many` calls:

```ruby
has_many :deliveries, class_name: "WebhookDelivery", dependent: :delete_all
has_many :headers, class_name: "WebhookHeader", dependent: :delete_all
```